R/utils.R

```diff
@@ -76,3 +76,144 @@ ensure_data.table <- function(data) {
   }
   return(data)
 }
+
+
+#' @title Check whether an input is an atomic vector of mode 'numeric'
+#'
+#' @description Helper function to check whether an input is a numeric vector.
+#' @param x input to check
+#' @inheritDotParams checkmate::check_numeric
+#' @importFrom checkmate check_atomic_vector check_numeric
+#' @inherit document_check_functions return
+#' @keywords internal_input_check
+check_numeric_vector <- function(x, ...) {
+  # check functions must return TRUE on success
+  # and a custom error message otherwise
+  numeric <- check_numeric(x, ...)
+  vector <- check_atomic_vector(x)
+  if (!isTRUE(numeric)) {
+    return(numeric)
+  } else if (!isTRUE(vector)) {
+    return(vector)
+  }
+  return(TRUE)
+}
+
+
+#' Get forecast type from forecast object
+#' @inheritParams score
+#' @return
+#' Character vector of length one with the forecast type.
+#' @keywords internal_input_check
+get_forecast_type <- function(forecast) {
```

Comment:
I think I would put these forecast type tools in their own file as they share a clear use case and I think users implementing their own class might want to be aware of them?